### ISSUE TITLE
bbb-html5-backend and bbb-html5-frontend are failing to start after Upgrade 2.3.15 -> 2.4.0 because of pending mongodb start

### ISSUE THREAD

**Issue by olwe0002** [bigbluebutton/bigbluebutton#13981]
**Describe the bug**
After upgrading from 2.3.15 to 2.4.0, the bbb-html5-backend@1, bbb-html5-backend@2, bbb-html5-frontend@1, bbb-html5-frontend@2 services fail to start.

The reason is a race condition with mongodb:

```
systemctl status bbb-html5-backend@1
● bbb-html5-backend@2.service - BigBlueButton HTML5 service, backend instance 2
   Loaded: loaded (/usr/lib/systemd/system/bbb-html5-backend@.service; disabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Thu 2021-12-23 15:26:49 CET; 10h ago
  Process: 20993 ExecStart=/usr/share/meteor/bundle/systemd_start.sh 2 $BACKEND_NODEJS_ROLE (code=exited, status=1/FAILURE)
 Main PID: 20993 (code=exited, status=1/FAILURE)

Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: Initializing replicaset
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: MongoDB shell version v4.2.17
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: connecting to: mongodb://127.0.1.1:27017/test?compressors=disabled&gssapiServiceName=mongodb
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: 2021-12-23T15:26:49.171+0100 E  QUERY    [js] Error: couldn't connect to server 127.0.1.1:27017, connecti
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: connect@src/mongo/shell/mongo.js:353:17
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: @(connect):2:6
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: 2021-12-23T15:26:49.207+0100 F  -        [main] exception: connect failed
Dec 23 15:26:49 vm-staging-big-blue-button systemd_start.sh[20993]: 2021-12-23T15:26:49.207+0100 E  -        [main] exiting with code 1
Dec 23 15:26:49 vm-staging-big-blue-button systemd[1]: bbb-html5-backend@2.service: Main process exited, code=exited, status=1/FAILURE
Dec 23 15:26:49 vm-staging-big-blue-button systemd[1]: bbb-html5-backend@2.service: Failed with result 'exit-code'.
```

the strange thing is each second `bbb-conf --restart`, it works. each second `bbb-conf --restart`, it fails. If I manually start mongod service, wait some time and then start manually bbb-html5-backend@1 service, it is running fine.

**To Reproduce**
1. upgrade from 2.3.15 to 2.4.0
2. `bbb-conf --status`
3. `bbb-conf --restart`
4. `bbb-conf --status`
5.  `bbb-conf --restart`
6. `bbb-conf --status`

**Expected behavior**
all processes are running on each restart

**Actual behavior**
```
bbb-conf --status
nginx —————————————————► [✔ - active]
freeswitch ————————————► [✔ - active]
redis-server ——————————► [✔ - active]
bbb-apps-akka —————————► [✔ - active]
bbb-fsesl-akka ————————► [✔ - active]
mongod ————————————————► [✔ - active]
bbb-html5 —————————————► [✔ - active]
bbb-webrtc-sfu ————————► [✔ - active]
kurento-media-server ——► [✔ - active]
kurento-media-server-8888  [✔ - active]
kurento-media-server-8889  [✔ - active]
kurento-media-server-8890  [✔ - active]
bbb-html5-backend@1 ———► [✘ - failed]
bbb-html5-backend@2 ———► [✘ - failed]
bbb-html5-frontend@1 ——► [✘ - failed]
bbb-html5-frontend@2 ——► [✘ - failed]
etherpad ——————————————► [✔ - active]
bbb-web ———————————————► [✔ - active]
```

**BBB version:**
 2.4.0

**Comment by antobinary** [comment]
Hi @olwe0002 
Thanks for reporting this!

> the strange thing is each second bbb-conf --restart, it works. each second bbb-conf --restart, it fails. If I manually start mongod service, wait some time and then start manually bbb-html5-backend@1 service, it is running fine.

This sounds odd indeed. We'll take a deeper look. It seems similar to https://github.com/bigbluebutton/bigbluebutton/issues/11580

**Comment by andreasotto** [comment]
Confirmed same issue here. Absolutely fresh install on ubuntu 18.04 LTS exactly like specified.

+1

**Comment by zhem0004** [comment]
I have tried to reproduce this issue on 2.5 and it is not present there. 
I tried to run the instructions to reproduce it several dozens of times, tried to increase CPU and internet usage as much as possible, also tried to run updates at the same time. However, this situation did not occur.

**Comment by prlanzarin** [comment]
> I have tried to reproduce this issue on 2.5 and it is not present there. I tried to run the instructions to reproduce it several dozens of times, tried to increase CPU and internet usage as much as possible, also tried to run updates at the same time. However, this situation did not occur.

FYI we had this same issue in 2.4, although it was a fairly rare thing to happen. Our "fix" was to backport https://github.com/bigbluebutton/bigbluebutton/pull/13527 (and its subsequent PR) from 2.5 to 2.4.

**Comment by antobinary** [comment]
Thanks for this comment!

The changes @prlanzarin is referring to were backported to 2.4 for the 2.4.1 release judging by the release notes https://github.com/bigbluebutton/bigbluebutton/releases/tag/v2.4.1

If this is the "fix" that resolved the issue for @prlanzarin , then this is "fixed" as of BBB 2.4.1 and on all BBB 2.5.x [dev] releases so far

**Comment by ho30hero** [comment]
Hi
I have the same issue in 2.5.2

```
nginx —————————————————► [✔ - active]
freeswitch ————————————► [✔ - active]
redis-server ——————————► [✔ - active]
bbb-apps-akka —————————► [✔ - active]
bbb-fsesl-akka ————————► [✔ - active]
tomcat9 ———————————————► [✔ - active]
mongod ————————————————► [✔ - active]
bbb-html5 —————————————► [✔ - active]
bbb-webrtc-sfu ————————► [✔ - active]
kurento-media-server ——► [✔ - active]
bbb-html5-backend@1 ———► [✘ - failed]
bbb-html5-backend@2 ———► [✘ - failed]
bbb-html5-frontend@1 ——► [✘ - failed]
bbb-html5-frontend@2 ——► [✘ - failed]
etherpad ——————————————► [✔ - active]
bbb-web ———————————————► [✔ - active]
bbb-pads ——————————————► [✔ - active]

```
